scvi/model/_scvi.py

```diff
@@ -290,7 +292,28 @@ def to_latent_mode(
             Key to use in `adata.obsm` where the latent qzm params are stored
         use_latent_qzv_key
             Key to use in `adata.obsm` where the latent qzv params are stored
+        minify_adata
+            If True, the anndata object associated with the model is replaced
+            by a new object with only the latent representation and metadata.
+
+        Notes
+        -----
+        A new, minimal adata object is associated as `model.adata` after running
+        this method with `minify_adata=True`. This adata does not contain any of
+        the original count data, but instead contains the latent representation
+        of the original data and metadata.
         """
-        self._get_latent_adata_from_adata(mode, use_latent_qzm_key, use_latent_qzv_key)
-        self.adata_manager.register_new_fields(self.__class__._get_latent_fields(mode))
+        if minify_adata:
+            bdata = self._get_reduced_adata(mode)
+        else:
+            bdata = self.adata
```

Comment:
Why should we allow a latent mode where the data is not dehydrated?

Reply:
for cached inference without altering data

Reply:
I dont think we should mix the two notions under the umbrella of latent data.
The codebase (and developer interface) currently assumes that latent == no count data. If you want to shove both states under the same notion (which is feasible but would probably not something I would do) then at a minimum the LatentDataType or other data "tag" needs to capture that. Unless you would want call sites to check if the data is "real latent" or "half latent".
What I'd do for this is I'd probably use LatentDataType to include whether the full data is present as well or not.

Reply:
good point, should be fixed now